codegen/type_converter.go

```diff
@@ -55,6 +55,25 @@ func (c *TypeConverter) appendf(format string, parts ...interface{}) {
 	c.Lines = append(c.Lines, line)
 }
 
+func (c *TypeConverter) assignWithOverride(
+	indent string,
+	toIdentifier string,
+	typeName string,
+	fromIdentifier string,
+	overriddenIdentifier string,
+) {
+	if overriddenIdentifier == "" {
+		c.append(indent, toIdentifier, " = ", typeName, "(", fromIdentifier, ")")
+		return
+	}
+
+	// TODO: Verify all intermediate objects are not nil
+	c.append(indent, toIdentifier, " = ", typeName, "(", overriddenIdentifier, ")")
```

Comment:
Why do we assign from both fields ? 

It looks like a transform should only support the transformed field not the original field.